qa/tasks/cephfs/test_snap_schedules.py

```diff
@@ -0,0 +1,193 @@
+import os
+import json
+import time
+import errno
+import random
+import logging
+import collections
+
+from tasks.cephfs.cephfs_test_case import CephFSTestCase
+from teuthology.exceptions import CommandFailedError
+
+log = logging.getLogger(__name__)
+
+def extract_schedule_and_retention_spec(spec=[]):
+    schedule = set([s[0] for s in spec])
+    retention = set([s[1] for s in spec])
+    return (schedule, retention)
+
+def seconds_upto_next_schedule(time_from, timo):
```

Comment:
minor nit: this routine would not be required w/ @jan--f latest changes to PR #29489